src/functions/template-tags/date.php

```diff
@@ -96,76 +96,29 @@ function tribe_get_end_time( $event = null, $dateFormat = '', $timezone = null )
 	}
 }
 
-if ( ! function_exists( 'tribe_get_start_date' ) ) {
+if ( ! function_exists( 'tribe_get_display_end_date' ) ) {
 	/**
-	 * Start Date
+	 * End Date formatted for display
 	 *
-	 * Returns the event start date and time
+	 * Returns the event end date that observes the end of day cutoff
 	 *
 	 * @category Events
-	 * @param int    $event       (optional)
-	 * @param bool   $displayTime If true shows date and time, if false only shows date
-	 * @param string $dateFormat  Allows date and time formating using standard php syntax (http://php.net/manual/en/function.date.php)
-	 * @param string $timezone    Timezone in which to present the date/time (or default behaviour if not set)
-	 * @return string|null Date
-	 */
-	function tribe_get_start_date( $event = null, $displayTime = true, $dateFormat = '', $timezone = null ) {
-		if ( is_null( $event ) ) {
-			global $post;
-			$event = $post;
-		}
-
-		if ( is_numeric( $event ) ) {
-			$event = get_post( $event );
-		}
-
-		if ( ! is_object( $event ) ) {
-			return '';
-		}
-
-		if ( tribe_event_is_all_day( $event ) ) {
-			$displayTime = false;
-		}
-
-		$start_date = Tribe__Events__Timezones::event_start_timestamp( $event->ID, $timezone );
-		return tribe_format_date( $start_date, $displayTime, $dateFormat );
-	}
-}
-
-if ( ! function_exists( 'tribe_get_end_date' ) ) {
-	/**
-	 * End Date
-	 *
-	 * Returns the event end date
-	 *
-	 * @category Events
-	 * @param int    $event       (optional)
-	 * @param bool   $displayTime If true shows date and time, if false only shows date
-	 * @param string $dateFormat  Allows date and time formating using standard php syntax (http://php.net/manual/en/function.date.php)
-	 * @param string $timezone    Timezone in which to present the date/time (or default behaviour if not set)
+	 * @param int    $event        (optional)
+	 * @param bool   $display_time If true shows date and time, if false only shows date
+	 * @param string $date_format  Allows date and time formating using standard php syntax (http://php.net/manual/en/function.date.php)
+	 * @param string $timezone     Timezone in which to present the date/time (or default behaviour if not set)
 	 *
 	 * @return string|null Date
 	 */
-	function tribe_get_end_date( $event = null, $displayTime = true, $dateFormat = '', $timezone = null ) {
-		if ( is_null( $event ) ) {
-			global $post;
-			$event = $post;
-		}
-
-		if ( is_numeric( $event ) ) {
-			$event = get_post( $event );
-		}
+	function tribe_get_display_end_date( $event = null, $display_time = true, $date_format = '', $timezone = null ) {
+		$end_date = tribe_get_end_date( $event, true, 'U', $timezone );
+		$beginning_of_day = tribe_beginning_of_day( date( Tribe__Date_Utils::DBDATETIMEFORMAT, $end_date ) );
 
-		if ( ! is_object( $event ) ) {
-			return '';
+		if ( tribe_event_is_multiday( $event ) && $end_date < strtotime( $beginning_of_day ) ) {
```

Comment:
The follow-up to this question is in Slack :D https://tribe.slack.com/archives/products-dev/p1449609622002735

The short answer is, no. We don't need to make that adjustment for the bug as described.